catwalk/tasks/metaicl.py

```diff
@@ -87,17 +87,17 @@ def instance_as_rank_classification(
             fewshot_instances = []
         prefix = ""
         for fewshot_instance in fewshot_instances:
-            as_rc = self.instance_as_rank_classification(fewshot_instance)
+            as_rc = self.instance_as_rank_classification(fewshot_instance, continuation_seperator=continuation_seperator, example_seperator=example_seperator)
             if as_rc.correct_choice is None:
                 raise ValueError("Could not determine correct choice in ranked classification instance.")
             correct_choice = as_rc.choices[as_rc.correct_choice]
-            prefix += f"{correct_choice[0].strip()}{continuation_seperator}{correct_choice[1].strip()}{example_seperator}"
+            prefix += correct_choice[0] + correct_choice[1] + example_seperator
```

Comment:
There is no separator between the choices?

Reply:
The separator is already included when the fewshot_instances go into `instance_as_rank_classification()`themselves. The old code would strip that and reapply the separator here but that was problematic for examples that have leading or trailing whitespace that needs to be kept to reproduce metrics.

Reply:
That's fine. But out of curiosity, why does the white space still matter after tokenization?

Reply:
Oh yes I had the same thought! It turns out that gpt2 tokenization has an explicit trailing space token (`220` I think). It has to use it when a space is at the end of an input and thus can't be attached to the next token.